configure.ac

```diff
@@ -178,6 +178,7 @@ case "${host_cpu}" in
       AC_DEFINE([HAVE_NEON], [1], [Enable NEON instructions])
       NEON_CXXFLAGS="-mfpu=neon"
       AC_SUBST([NEON_CXXFLAGS])
+      check_for_neon=1
```

Comment:
I suggest to include the code from below here. Then there is no need for `check_for_neon`.

Reply:
Which code do you mean specifically?

Reply:
Lines 197–202.

Reply:
Line 196 checks for `check_for_neon`.  The point of this variable is to only run the NEON checks on platforms where NEON could possibly be available.  You [specifically requested](https://github.com/tesseract-ocr/tesseract/pull/3782#discussion_r841081250) that the test be implemented that way.

Reply:
Right. And now I suggest to move lines 197–202 to replace line 181 in your commit. That also runs the NEON checks only when needed, but simplifies and reduces the code for that.

Reply:
There was some sort of reason why I did it the way I did it.  I can try to check if it works the way you indicate, but it has been almost two months since I last touched this changeset and it will be difficult for me to re-run all the tests and ensure it still works correctly.  It would be a lot easier for me just to commit it as is.

Reply:
Then please add at least all other suggested changes.